crates/fluvio-spu-schema/src/server/api.rs

```diff
@@ -67,3 +67,27 @@ impl ApiMessage for SpuServerRequest {
         }
     }
 }
+
+impl SpuServerRequest {
+    pub fn correlation_id(&self) -> i32 {
+        match self {
```

Comment:
why is this needed?

Reply:
I added this to make it easy to get to correlation_id and client_id and add that information to the span in the main SPU request handler function: https://github.com/infinyon/fluvio/pull/1532/files#diff-2277166e98453fe79d4a9c077d19f29f5e51fbb911900f3671aea595cbf572d3R75-R76

These are actually two critical pieces of information that should be included at the top-level request span, and should help debugging ability significantly, e.g. you can filter for a specific correlation_id when looking at logs to see the flow of events.

Reply:
Here is current trace:
```
 handle_request{host="0.0.0.0:9010"}:respond
{socket=Socket(13) _connection=ConnectInfo { host: "0.0.0.0:9010", peer: "127.0.0.1:48528" }}:s
```

We should generalize, 
* socket
* port
* peer
* client_id
* correlation_id

This should be done highest possible to handle any API.

Reply:
I agree, let's look at how we can do that. I just broke this PR into two pieces because something in the second half broke the build, so let's come back to this when I open the next PR with that code